File: src/game/ActionEvent.cpp
#include "ActionEvent.h"

#include "SDL/SDL_types.h"

void ActionEvent::stop() {
}

ActionEvent::ActionEvent(Uint32 p) {
	delay = p;
}

void ActionEvent::addCurtick(Uint32 curtick) {
	prio = delay + curtick;
}

void ActionEvent::run() { }

ActionEvent::~ActionEvent() {
}

void ActionEvent::setDelay(Uint32 p) {
	delay = p;
}

Uint32 ActionEvent::getPrio() {
	return prio;
}

File: java/src/openra/server/ActionEvent.java
package openra.server;

public class ActionEvent {
	
	long delay;
	long prio;
	private ActionEventQueue aequeue;

	void stop()
	{
	}

	ActionEvent(long p)
	{
	    delay = p;
	}

	void addCurtick(long curtick)
	{
	    prio = delay + curtick;
	}

	void run()
	{
	}

	

	void setDelay(long p)
	{
	    delay = p;
	}

	long getPrio()
	{
	    return prio;
	}

	public void setActionEventQueue(ActionEventQueue actionEventQueue) {
		this.aequeue = actionEventQueue;
	}

	/**
	 * @return the aequeue
	 */
	public ActionEventQueue getAequeue() {
		return aequeue;
	}

	
}
